lambdas/FetchJobsDataFiltered.py

```diff
@@ -0,0 +1,138 @@
+    conditions = []
+    values = []
+
+    # Check if 'postedWithin' is present and not empty
+    if event.get('postedWithin'):
+        target_date = get_date_days_ago(days=event['postedWithin'])
+        conditions.append("date_posted >= %s")
+        values.append(target_date)
+
+    # Check if 'location' is present and not empty
+    if event.get('location'):
+        location_param = standardize_location(event['location'])
+        province_full = PROVINCE_MAPPING.get(location_param, location_param)
+        location_condition = "(REPLACE(LOWER(location), ' ', '') ILIKE %s OR REPLACE(LOWER(location), ' ', '') ILIKE %s)"
+        conditions.append(location_condition)
+        values.extend([f"%{location_param}%", f"%{province_full}%"])
+
+    # Check if 'title' is present and not empty
+    if event.get('title'):
+        title_param = event['title'].replace("-", " ")
+        title_condition = "title ILIKE %s"
+        conditions.append(title_condition)
+        values.append(f"%{title_param}%")
+
+    query = f"{base_query} WHERE {' AND '.join(conditions)} ORDER BY date_posted DESC" if conditions else base_query
+    return query, values
+
+
+def execute_query(cursor, query: str, values: List[Any]) -> List[Dict[str, Any]]:
+    """Executes the given SQL query and returns the result."""
+    try:
+        cursor.execute(query, values)
+        return cursor.fetchall()
+    except Exception as e:
+        logging.error("Query execution failed: %s", e)
+        raise
+
+def lambda_handler(event, context):
+    """Entry point for the AWS Lambda function."""
+    with get_db_connection() as conn:
+        with conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor) as cursor:
+            query, values = build_query(event)
+            results = execute_query(cursor, query, values)
+            json_results = json.dumps(results, default=str)
+
+    return {
+        'statusCode': 200,
+        'body': json_results,
+        'records_fetched': len(results)
+    }
+
+# Local development testing...
+if __name__ == "__main__":
+    # for local development, run the lambda function
+    logging.basicConfig(level=logging.INFO)
+    
+    # Mock event for filtering data, all empty = get all jobs
+    mock_event = {'location': 'BC', 'postedWithin': '3', 'title': 'software-engineer'}
+    result = lambda_handler(mock_event, None)
+
+    # Extracting the statusCode, body, and records_fetched from the result
+    status_code = result['statusCode']
+    records_fetched = result['records_fetched']
+    body = json.loads(result['body'])
+
+    # Print statusCode, one example from body (if available), and records_fetched
+    print(f"Status Code: {status_code}")
+    # print(f"Example Record: {body[0] if body else 'No records found'}")
+    print(f"Example Record: {body if body else 'No records found'}")
+    
+    print(f"Total Records Fetched: {records_fetched}")
```

Comment:
Why are we printing these btw? for development debugging? but these lambda functions are triggered on AWS right? I'm not quite clear on this. 

Maybe remove these if not applicable

Reply:
These won't run on production lambda, lambda will only execute the lambda_handler() function, so these print statements will only be run when running the script under local development. This might not be the best practice, I think people use AWS SAM for local development, I will look more into that but we could keep this for now